### DISCUSSION TITLE
Live state visualisation

### DISCUSSION THREAD

**Discussion by truthz03** [elsa-workflows/elsa-core#3550] [Q&A]
Is it possible to visualize the current state of a workflow?
I know that every workflow execution will be shown inside the "Workflow Instances" page but the state of the single activities will only be shown if the workflow has finished or failed.
Is it possible to get activity state updates while the workflow is running (maybe per SignalR)?
This would help to analyze long running worklows.

Thanks
Thomas

**Comment by sjd2021** [comment]
"the state of the single activities will only be shown if the workflow has finished or failed."

I could be very wrong, but I don't believe this is the experience I am having. We use blocking activities that await user input, and (it's been a while) I believe I was able to see the current state of the workflow instance's execution when it was in a suspended state.

As for signalR updates, Elsa was thankfully constructed with Mediatr so I think that if you open up the repository you might find a number of events that you can subscribe to in order to add your own notifications. It's a really great library for this reason.

**Reply by truthz03** [reply]
With my test I'm not able to see which activity is currently blocking the workflow instance.

At the "Workflow Instances" page is saw that there is a Running instance
![image](https://user-images.githubusercontent.com/14071920/208644320-ac4d35c9-4d7d-4620-b31f-e2ef23f452e8.png)

If I click this instance none of the steps shows as executed.
The "Delay http request" activity makes an api call which blocks 30sec before it returns 
but within these 30sec the "HTTP Endpoint" activity doesn't show the green icon.
![image](https://user-images.githubusercontent.com/14071920/208644248-01ed834a-c17e-4966-88f4-1f3181b76eb5.png)

After the 30sec the workflow finishes and then everything is green.
![image](https://user-images.githubusercontent.com/14071920/208645206-03452f67-267a-4e27-a860-da48c93c8ae3.png)

**Reply by mubinahmed** [reply]
@truthz03 Did you get solution for this?

**Reply by truthz03** [reply]
No, I had no time until yet to search for a solution
